### PR TITLE
fix(TDI-40195): Case sensitivity SnowflakeInput

### PR DESCRIPTION
**Pull request by olhava**
**What is the current behavior?** (You can also link to an open issue here)
Snowflake Input with condition mode crashes with exception if columns/table names are in lower case

**What is the new behavior?**
Option 'Convert columns and table to uppercase' is added to advanced settings, true by default.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1830/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1830/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aovarnavina%2FTDI-40195_snowflake_input_lowercase)

File: components/components-snowflake/src/main/java/org/talend/components/snowflake/runtime/SnowflakeReader.java [hunk 1]
```diff
@@ -92,16 +92,19 @@ protected String getQueryString() throws IOException {
...
         StringBuilder sb = new StringBuilder();
+        boolean isUpperCase = properties.convertColumnsAndTableToUppercase.getValue();
         sb.append("select "); //$NON-NLS-1$
         int count = 0;
         for (Schema.Field se : getSchema().getFields()) {
             if (count++ > 0) {
                 sb.append(", "); //$NON-NLS-1$
             }
-            sb.append(se.name());
+            String fName = isUpperCase ? se.name() : "\"" + se.name() + "\"";
```
Comment (mbasiuk-talend) [inline components/components-snowflake/src/main/java/org/talend/components/snowflake/runtime/SnowflakeReader.java]: Can we prevent creation of unneccessary Strings with "\" " surrounded. On my opinion it's better to use condition for appending " \"" before and after name. Dmytro, what do you think?

Comment (dmytro-sylaiev) [reply components/components-snowflake/src/main/java/org/talend/components/snowflake/runtime/SnowflakeReader.java]: I'd prefer to use `org.apache.commons.lang3.StringUtils.wrap(se.name(), "\"");` here

File: components/components-snowflake/src/main/java/org/talend/components/snowflake/runtime/SnowflakeReader.java [hunk 2]
```diff
@@ -92,16 +92,19 @@ protected String getQueryString() throws IOException {
...
+            sb.append(fName);
         }
         sb.append(" from "); //$NON-NLS-1$
-        sb.append(properties.table.tableName.getStringValue());
+        String tableName = isUpperCase ? properties.getTableName() : "\"" + properties.getTableName() + "\"";
+        sb.append(tableName);
```
Comment (dmytro-sylaiev) [inline components/components-snowflake/src/main/java/org/talend/components/snowflake/runtime/SnowflakeReader.java]: Same here

File: components/components-snowflake/src/main/java/org/talend/components/snowflake/tsnowflakeinput/TSnowflakeInputProperties.java [hunk 3]
```diff
@@ -57,6 +59,7 @@ public void setupProperties() {
         super.setupProperties();
         manualQuery.setValue(false);
         query.setTaggedValue(ComponentConstants.LINE_SEPARATOR_REPLACED_TO, " ");
+        convertColumnsAndTableToUppercase.setValue(true);
```
Comment (dmytro-sylaiev) [inline components/components-snowflake/src/main/java/org/talend/components/snowflake/tsnowflakeinput/TSnowflakeInputProperties.java]: Seems you already set `true` as initial value in Property<Boolean> constructor.
Why do you need this line either?

File: components/components-snowflake/src/test/java/org/talend/components/snowflake/runtime/SnowflakeReaderTest.java [hunk 4]
```diff
@@ -148,6 +150,25 @@ public void testGetManualQuery() throws Exception {
...
+        Mockito.when(resultSetMock.getMetaData()).thenReturn(Mockito.mock(ResultSetMetaData.class));
+        Mockito.when(resultSetMock.getMetaData().getColumnCount()).thenReturn(2);
+        Mockito.when(resultSetMock.getString(0)).thenReturn("row1field", "row1column");
+        Mockito.when((snowflakeSourceMock).createConnection(runtimeContainerMock)).thenReturn(connectionMock);
+        Mockito.when(connectionMock.createStatement()).thenReturn(statementMock);
+        Mockito.when(statementMock.executeQuery(TEST_LOWERCASE_NAMES_QUERY)).thenReturn(resultSetMock);
+
+        snowflakeReader.start();
+        snowflakeReader.getCurrent();
+    }
```
Comment (dmytro-sylaiev) [inline components/components-snowflake/src/test/java/org/talend/components/snowflake/runtime/SnowflakeReaderTest.java]: It is a bad practice to have a unit test without any assertions.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1846/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1846/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aovarnavina%2FTDI-40195_snowflake_input_lowercase)

**Pull request merged**